src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx

```diff
@@ -107,21 +124,34 @@ export const ProfilePictureInput: React.FunctionComponent<ProfilePictureInputPro
     ? customAvatars
     : DEFAULT_AVATARS;
 
-  const avatarImages = useMemo(() => {
-    return defaultAvatars.map((avatar, index) => (
+  const getRenderedAvatarImageComponent = useCallback(
+    (avatarSrc: string, key: string | number) => (
       <div
-        key={`${avatar}-${index}`}
+        key={`${avatarSrc}-${key}`}
         className="profile-picture-preview-container"
-        onClick={() => uploadDefaultAvatar(avatar)}
+        onClick={() => uploadDefaultAvatar(avatarSrc)}
       >
         <img
-          src={avatar}
+          src={avatarSrc}
           className="profile-icon profile-picture-preview"
-          alt={`default avatar ${index}`}
+          alt={`default avatar ${key}`}
         />
       </div>
-    ));
-  }, [defaultAvatars, uploadDefaultAvatar]);
+    ),
+    [uploadDefaultAvatar]
+  );
+
+  const gitHubImage = useMemo(
+    () =>
+      githubImageSrc &&
+      getRenderedAvatarImageComponent(githubImageSrc, "github-avatar"),
+    [githubImageSrc, getRenderedAvatarImageComponent]
+  );
+
+  const avatarImages = useMemo(
+    () => defaultAvatars.map(getRenderedAvatarImageComponent),
+    [defaultAvatars, getRenderedAvatarImageComponent]
+  );
```

Comment:
Why not combine this together in a single `useMemo`, instead of splitting it across 3 (`getRenderedAvatarImageComponent `, `gitHubImage`, `avatarImages`)?

Reply:
Fixes in https://github.com/sparkletown/sparkle/pull/1667/commits/8c56ac87d7d1fd51d6b29db5df212b7a3a948efc